lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -327,6 +327,10 @@ public function delete_item( $request ) {
 			return new WP_Error( 'rest_user_cannot_delete_post', __( 'Sorry, you are not allowed to delete this post.' ), array( 'status' => 401 ) );
 		}
 
+		$mock_request = new WP_REST_Request( 'GET', rest_url( 'wp/v2/' . $this->get_post_type_base( $this->post_type ) . '/' . $post->ID ) );
+		$mock_request->set_param( 'context', 'edit' );
```

Comment:
`$mock_request` isn't really the right name, since this is a real request and "mock" has a specific meaning. `$get_request`?